### PR TITLE
feat(ics08-movement): further movement verification

### PR DESCRIPTION
**Pull request by aeryz**

### CODE REVIEW

File: lib/aptos-verifier/src/lib.rs [hunk 1]
```diff
@@ -70,25 +71,19 @@ pub fn verify_tx_state(
...
-    buf.write_all(&key).unwrap();
-
-    let hash = Sha3_256::new()
-        .chain_update(Sha3_256::new().chain_update("APTOS::StateKey").finalize())
-        .chain_update(&buf)
-        .finalize()
-        .into();
+    let Some(proof_leaf) = proof.leaf.clone() else {
+        return Err(StorageVerificationError::ExpectedMembershipVerification.into());
+    };
```
Comment (benluelo) [inline lib/aptos-verifier/src/lib.rs:79]: why clone?

File: voyager/modules/consensus/movement/src/main.rs [hunk 2]
```diff
@@ -221,12 +245,45 @@ impl ConsensusModuleServer<ModuleData, ModuleCall, ModuleCallback> for ModuleSer
...
+            .unwrap()
+            .to_owned();
+
         Ok(serde_json::to_value(movement::client_state::ClientState {
             chain_id: self.ctx.chain_id.to_string(),
             l1_client_id: self.ctx.l1_client_id.clone(),
             l1_contract_address: self.ctx.l1_settlement_address,
             l2_contract_address: self.ctx.ibc_handler_address,
-            table_handle: AccountAddress(Default::default()),
+            table_handle: AccountAddress(Hash::new(
```
Comment (benluelo) [inline voyager/modules/consensus/movement/src/main.rs:284]:
```suggestion
            table_handle: AccountAddress(H256::new(
```

Comment (aeryz) [reply voyager/modules/consensus/movement/src/main.rs:284]: `AccountAddress` has the type `Hash` as the inner type im confused

File: lib/aptos-verifier/src/lib.rs (hunk 1 above)
Comment (aeryz) [reply lib/aptos-verifier/src/lib.rs:79]: bc im stupid

File: voyager/modules/consensus/movement/src/main.rs (hunk 2 above)
Comment (benluelo) [reply voyager/modules/consensus/movement/src/main.rs:284]: sorry forgot my other PR wasn't merged yet, disregard

**Review by benluelo — APPROVED** [review]

**Review by benluelo — APPROVED** [review]

**Pull request merged**